### DISCUSSION TITLE
Collision between two signed-distance-field objects

### DISCUSSION THREAD

**Discussion by rstrudel** [bulletphysics/bullet3#3577] [General]
Hi,
The signed-distance-field feature seems great, I have tried to use it and got some problems on the way. Collision between a static mesh using CDF and non-static meshes using VHACD as collision shape works well (as in the example with duck, spheres and a concave box).

However it seems that collision between two meshes using CDF is not handled . Here is a simple piece of code reproducing the error using the `concave_box.urdf` file and setting the mass to 1 :

```python
import pybullet as pb
from time import sleep

pb.connect(pb.GUI)
pb.setPhysicsEngineParameter(numSolverIterations=10)
pb.setTimeStep(1./120.)
pb.setGravity(0, 0, -10)

pb.loadURDF("toys/concave_box.urdf",[0,0,2])
pb.loadURDF("toys/concave_box.urdf",[0,0,0],useFixedBase=True)

while 1:
    sleep(0.01) # Time in seconds.
    pb.stepSimulation()
```

The top `concave_box` just go through the fixed one and no collisions occur.

**Comment by erwincoumans** [comment]
This is not implemented at the moment. We would need to provide some way get surface points on one SDF against the other. I suspect there won't be anyone working on this, so contributions are welcome.

**Comment by SirNate0** [comment]
I know nothing about the feature (in fact, I didn't know Bullet even offered signed distance field objects until I came across this issue), but wouldn't something like checking for the intersection of the two signed distance fields be simple enough, something like just computing the min between the two of them and sampling over the regions within them that might intersect (based on a bounding box or convex hull or something)? As I said, though, I'm not familiar with the feature at all, and I don't know if that approach would be suitable for per-step calculations nor whether it would be amenable to some sort of up-front cacheable calculation like finding surface points could be.

**Comment by benediktkreis** [comment]
Are there any updates on this feature?

I'm trying to simulate several dynamic objects defined by concave meshes. Unfortunately, the standard VHACD collision shape is too coarse grained for my application. Using the original C++ VHACD implementation results in more fine grained collision shapes. But these shapes extremely slow down the PyBullet simulation. I hoped to avoid the shortcomings of VHACD by using SDF. Otherwise, I'm forced to use another simulator that has this feature.

**Reply by RafaelsNeurons** [reply]
@benediktkreis, do you know a physics engine which is able to do this? (preferably executable in Python and Windows?)

**Reply by benediktkreis** [reply]
@RafaelsNeurons MuJoCo is supporting it since October last year ([see changelog](https://mujoco.readthedocs.io/en/3.1.2/changelog.html#version-3-0-0-october-18-2023)). I haven't tested it myself though.

**Reply by RafaelsNeurons** [reply]
Thanks I will try that! Hope it works with the Python Bindings 😅
